bug:

### Describe the bug

We observe a panic in kcp-stable:

```
I0920 06:18:45.080532       1 httplog.go:131] "HTTP" verb="LIST" URI="/clusters/root:users:xr:vd:rh-sso-hongweiliu:integration-ws/api/v1/namespaces" latency="30.807977ms" userAgent="kcp/v1.24.3+kcp (linux/amd64) kubernetes/7e4e8d8/kcp-workspace-deletion-controller" audit-ID="4114112e-8f37-4b7f-9473-7a27fd3fec2a" srcIP="[::1]:35884" apf_pl="e
xempt" apf_fs="exempt" apf_execution_time="30.636802ms" resp=200
I0920 06:18:45.080828       1 httplog.go:131] "HTTP" verb="LIST" URI="/apis/managed-gitops.redhat.com/v1alpha1/gitopsdeployments" latency="251.607µs" userAgent="kcp/v1.24.3+kcp (linux/amd64) kubernetes/7e4e8d8/kcp-apibinding-controller" audit-ID="f5495de4-e987-47f5-b663-74d171f62557" srcIP="[::1]:35884" apf_pl="exempt" apf_fs="exempt" apf_execution_time="80.766µs" resp=404
I0920 06:18:45.090080       1 httplog.go:131] "HTTP" verb="LIST" URI="/clusters/root:users:zu:bh:rh-sso-mnairn-hcg/api/v1/namespaces/kcp-system/resourcequotas" latency="10.586141ms" userAgent="kcp/v1.24.3+kcp (linux/amd64) kubernetes/7e4e8d8" audit-ID="f7c60c68-b480-4748-8a99-3516ab4ba91d" srcIP="[::1]:35884" apf_pl="exempt" apf_fs="exempt" apf_execution_time="10.37765ms" resp=200
E0920 06:18:45.090508       1 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 98168 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x3e2ec60?, 0x6428850})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apimachinery@v0.0.0-20220906141821-de523af4fb02/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0xc1644e4b68, 0x1, 0x2d56306?})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apimachinery@v0.0.0-20220906141821-de523af4fb02/pkg/util/runtime/runtime.go:49 +0x75
panic({0x3e2ec60, 0x6428850})
        /usr/local/go/src/runtime/panic.go:838 +0x207
github.com/kcp-dev/kcp/pkg/admission/webhook.(*WebhookDispatcher).getAPIBindingWorkspace(0xc000e7ee10?, {0x4923da0, 0xc122a198c0}, {{0xc15e8f3d80?, 0x3ec586716b1ea95b?}})
        /go/src/github.com/kcp-dev/kcp/pkg/admission/webhook/generic_webhook.go:105 +0x421
github.com/kcp-dev/kcp/pkg/admission/webhook.(*WebhookDispatcher).Dispatch(0xc0000b74f8, {0x490e428, 0xc125325260}, {0x4923da0, 0xc122a198c0}, {0x49121d8, 0xc121931a00})
        /go/src/github.com/kcp-dev/kcp/pkg/admission/webhook/generic_webhook.go:78 +0x10d
github.com/kcp-dev/kcp/pkg/admission/mutatingwebhook.(*Plugin).Admit(0xc000e7e430?, {0x490e428?, 0xc125325260?}, {0x4923da0?, 0xc122a198c0?}, {0x49121d8?, 0xc121931a00?})
        /go/src/github.com/kcp-dev/kcp/pkg/admission/mutatingwebhook/plugin.go:105 +0x3d
k8s.io/apiserver/pkg/admission/metrics.pluginHandlerWithMetrics.Admit({{0x48e1d20, 0xc0000b74f0}, 0xc000e7f5c0, {0xc000e7f5d0, 0x1, 0x1}}, {0x490e428, 0xc125325260}, {0x4923da0, 0xc122a198c0}, ...)
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/admission/metrics/metrics.go:95 +0xef
k8s.io/apiserver/pkg/admission.chainAdmissionHandler.Admit({0xc0000fe000?, 0x13, 0xc1644e52b0?}, {0x490e428, 0xc125325260}, {0x4923da0?, 0xc122a198c0?}, {0x49121d8, 0xc121931a00})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/admission/chain.go:37 +0x13d
k8s.io/apiserver/pkg/admission.(*reinvoker).Admit(0xffffffffffffffff?, {0x490e428, 0xc125325260}, {0x4923da0?, 0xc122a198c0?}, {0x49121d8, 0xc121931a00})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/admission/reinvocation.go:37 +0xad
k8s.io/apiserver/pkg/admission/metrics.pluginHandlerWithMetrics.Admit({{0x48e5ea0, 0xc000e7f740}, 0xc000e7f750, {0x0, 0x0, 0x0}}, {0x490e428, 0xc125325260}, {0x4923da0, 0xc122a198c0}, ...)
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/admission/metrics/metrics.go:95 +0xef
k8s.io/apiserver/pkg/admission.(*auditHandler).Admit(0xc0dbceb1d0, {0x490e428, 0xc125325260}, {0x4923da0, 0xc122a198c0}, {0x49121d8, 0xc121931a00})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/admission/audit.go:55 +0xde
k8s.io/apiserver/pkg/registry/rest.AdmissionToValidateObjectDeleteFunc.func1({0x490e428, 0xc125325260}, {0x48f1ef0?, 0xc16d208908?})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/registry/rest/delete.go:191 +0x611
k8s.io/apiserver/pkg/storage/etcd3.(*store).conditionalDelete(0xc118974ab0, {0x490e428, 0xc125325260}, {0xc18d83af00, 0xb5}, {0x48f1ef0, 0xc16d208900}, {0x3ee6740, 0xc16d208900, 0x199}, ...)
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/storage/etcd3/store.go:284 +0x3b1
k8s.io/apiserver/pkg/storage/etcd3.(*store).Delete(0xc118974ab0, {0x490e428, 0xc125325260}, {0xc1396b7550, 0xac}, {0x48f1ef0?, 0xc16d208900?}, 0x4?, 0x42a32db?, {0x48f1ef0, ...})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/storage/etcd3/store.go:221 +0x1e5
k8s.io/apiserver/pkg/storage/cacher.(*Cacher).Delete(0xc12252d200, {0x490e428, 0xc125325260}, {0xc1396b7550, 0xac}, {0x48f1ef0, 0xc16d208900}, 0x9?, 0xc16d208900?, {0x0, ...})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/storage/cacher/cacher.go:473 +0x113
k8s.io/apiserver/pkg/registry/generic/registry.(*DryRunnableStorage).Delete(0x4931e40?, {0x490e428?, 0xc125325260?}, {0xc1396b7550?, 0x0?}, {0x48f1ef0?, 0xc16d208900?}, 0xc1259596e0?, 0x14acaa9?, 0x0, ...)
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/registry/generic/registry/dryrun.go:56 +0x14b
k8s.io/apiserver/pkg/registry/generic/registry.(*Store).Delete(0xc11ec78280, {0x490e428, 0xc125325260}, {0xc111984ff0, 0x10}, 0x2d56865?, 0xc1259596e0)
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/registry/generic/registry/store.go:1102 +0x75b
k8s.io/apiserver/pkg/registry/generic/registry.(*Store).DeleteCollection.func2()
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/registry/generic/registry/store.go:1204 +0x257
created by k8s.io/apiserver/pkg/registry/generic/registry.(*Store).DeleteCollection
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/registry/generic/registry/store.go:1187 +0x31a
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3899901]

goroutine 98168 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0xc1644e4b68, 0x1, 0x2d56306?})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apimachinery@v0.0.0-20220906141821-de523af4fb02/pkg/util/runtime/runtime.go:56 +0xd8
panic({0x3e2ec60, 0x6428850})
        /usr/local/go/src/runtime/panic.go:838 +0x207
github.com/kcp-dev/kcp/pkg/admission/webhook.(*WebhookDispatcher).getAPIBindingWorkspace(0xc000e7ee10?, {0x4923da0, 0xc122a198c0}, {{0xc15e8f3d80?, 0x3ec586716b1ea95b?}})
        /go/src/github.com/kcp-dev/kcp/pkg/admission/webhook/generic_webhook.go:105 +0x421
github.com/kcp-dev/kcp/pkg/admission/webhook.(*WebhookDispatcher).Dispatch(0xc0000b74f8, {0x490e428, 0xc125325260}, {0x4923da0, 0xc122a198c0}, {0x49121d8, 0xc121931a00})
        /go/src/github.com/kcp-dev/kcp/pkg/admission/webhook/generic_webhook.go:78 +0x10d
github.com/kcp-dev/kcp/pkg/admission/mutatingwebhook.(*Plugin).Admit(0xc000e7e430?, {0x490e428?, 0xc125325260?}, {0x4923da0?, 0xc122a198c0?}, {0x49121d8?, 0xc121931a00?})
        /go/src/github.com/kcp-dev/kcp/pkg/admission/mutatingwebhook/plugin.go:105 +0x3d
k8s.io/apiserver/pkg/admission/metrics.pluginHandlerWithMetrics.Admit({{0x48e1d20, 0xc0000b74f0}, 0xc000e7f5c0, {0xc000e7f5d0, 0x1, 0x1}}, {0x490e428, 0xc125325260}, {0x4923da0, 0xc122a198c0}, ...)
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/admission/metrics/metrics.go:95 +0xef
k8s.io/apiserver/pkg/admission.chainAdmissionHandler.Admit({0xc0000fe000?, 0x13, 0xc1644e52b0?}, {0x490e428, 0xc125325260}, {0x4923da0?, 0xc122a198c0?}, {0x49121d8, 0xc121931a00})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/admission/chain.go:37 +0x13d
k8s.io/apiserver/pkg/admission.(*reinvoker).Admit(0xffffffffffffffff?, {0x490e428, 0xc125325260}, {0x4923da0?, 0xc122a198c0?}, {0x49121d8, 0xc121931a00})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/admission/reinvocation.go:37 +0xad
k8s.io/apiserver/pkg/admission/metrics.pluginHandlerWithMetrics.Admit({{0x48e5ea0, 0xc000e7f740}, 0xc000e7f750, {0x0, 0x0, 0x0}}, {0x490e428, 0xc125325260}, {0x4923da0, 0xc122a198c0}, ...)
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/admission/metrics/metrics.go:95 +0xef
k8s.io/apiserver/pkg/admission.(*auditHandler).Admit(0xc0dbceb1d0, {0x490e428, 0xc125325260}, {0x4923da0, 0xc122a198c0}, {0x49121d8, 0xc121931a00})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/admission/audit.go:55 +0xde
k8s.io/apiserver/pkg/registry/rest.AdmissionToValidateObjectDeleteFunc.func1({0x490e428, 0xc125325260}, {0x48f1ef0?, 0xc16d208908?})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/registry/rest/delete.go:191 +0x611
k8s.io/apiserver/pkg/storage/etcd3.(*store).conditionalDelete(0xc118974ab0, {0x490e428, 0xc125325260}, {0xc18d83af00, 0xb5}, {0x48f1ef0, 0xc16d208900}, {0x3ee6740, 0xc16d208900, 0x199}, ...)
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/storage/etcd3/store.go:284 +0x3b1
k8s.io/apiserver/pkg/storage/etcd3.(*store).Delete(0xc118974ab0, {0x490e428, 0xc125325260}, {0xc1396b7550, 0xac}, {0x48f1ef0?, 0xc16d208900?}, 0x4?, 0x42a32db?, {0x48f1ef0, ...})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/storage/etcd3/store.go:221 +0x1e5
k8s.io/apiserver/pkg/storage/cacher.(*Cacher).Delete(0xc12252d200, {0x490e428, 0xc125325260}, {0xc1396b7550, 0xac}, {0x48f1ef0, 0xc16d208900}, 0x9?, 0xc16d208900?, {0x0, ...})
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/storage/cacher/cacher.go:473 +0x113
k8s.io/apiserver/pkg/registry/generic/registry.(*DryRunnableStorage).Delete(0x4931e40?, {0x490e428?, 0xc125325260?}, {0xc1396b7550?, 0x0?}, {0x48f1ef0?, 0xc16d208900?}, 0xc1259596e0?, 0x14acaa9?, 0x0, ...)
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/registry/generic/registry/dryrun.go:56 +0x14b
k8s.io/apiserver/pkg/registry/generic/registry.(*Store).Delete(0xc11ec78280, {0x490e428, 0xc125325260}, {0xc111984ff0, 0x10}, 0x2d56865?, 0xc1259596e0)
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/registry/generic/registry/store.go:1102 +0x75b
k8s.io/apiserver/pkg/registry/generic/registry.(*Store).DeleteCollection.func2()
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/registry/generic/registry/store.go:1204 +0x257
created by k8s.io/apiserver/pkg/registry/generic/registry.(*Store).DeleteCollection
        /go/pkg/mod/github.com/kcp-dev/kubernetes/staging/src/k8s.io/apiserver@v0.0.0-20220906141821-de523af4fb02/pkg/registry/generic/registry/store.go:1187 +0x31a```

### Steps To Reproduce

observed on kcp-stable logs

### Expected Behaviour

no panic

### Additional Context

_No response_

--- a/pkg/admission/webhook/generic_webhook.go
+++ b/pkg/admission/webhook/generic_webhook.go
@@ -96,7 +96,7 @@ func (p *WebhookDispatcher) getAPIBindingWorkspace(attr admission.Attributes, cl
 	for _, obj := range objs {
 		apiBinding := obj.(*apisv1alpha1.APIBinding)
 		for _, br := range apiBinding.Status.BoundResources {
-			if apiBinding.Status.BoundAPIExport != nil && apiBinding.Status.BoundAPIExport.Workspace == nil {
+			if apiBinding.Status.BoundAPIExport == nil || apiBinding.Status.BoundAPIExport.Workspace == nil {
 				// this will never happen today. But as soon as we add other reference types (like exports), this log output will remind out of necessary work here.
 				klog.Errorf("APIBinding %s has no referenced workspace", clusterName, apiBinding.Name)
 				continue

